packages/vite/src/node/plugins/dynamicImportPolyfill.ts

```diff
@@ -5,14 +5,30 @@ import path from 'path'
 
 export const polyfillId = 'vite/dynamic-import-polyfill'
 
+/**
+ * fix #2918
+ * `path.posix.join` return a wrong path when config.base is a URL
+ * @param config
+ */
+function resolveModulePath(config: ResolvedConfig) {
+  const {
+    base,
+    command,
+    build: { assetsDir }
+  } = config
+  const isBuild = command === 'build'
```

Comment:
@hjxenjoy why do we need to check for isBuild here? If command is serve, it is going to be skipped, but I think it still makes send to have a properly formed URL, no?

Reply:
Yeah, it's ok. should I remove this condition and pr again?

Reply:
You don't need to do another, you can commit it to your local branch and push, the PR will be updated.

Reply:
ok!